### PR TITLE
make lfp processing work

### PR DESCRIPTION
**Pull request by a-zmz**

### CODE REVIEW

File: pixels/behaviours/base.py [hunk 1]
```diff
@@ -412,21 +412,45 @@ def process_lfp(self):
...
+            if output.exists():
+                continue
+
+            print("> Performing median subtraction across channels for each timepoint")
+            subtracted = signal.median_subtraction(data, axis=1)
+
             print(f"> Downsampling to {self.sample_rate} Hz")
-            data = signal.resample(data, orig_rate, self.sample_rate)
+            downsampled = signal.resample(subtracted, orig_rate, self.sample_rate)
+            sync_chan = downsampled[:, -1]
```
Comment (m-col) [inline pixels/behaviours/base.py:424]: Did you mean to pass this to `sync_data`?

Comment (a-zmz) [reply pixels/behaviours/base.py:424]: i kept the passing to `sync_data` bit as it is, did not touch it. But right now i do not use `sync_data`, i.e., i just comment it out when i use the function.

File: pixels/signal.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 from pixels import ioutils, PixelsError
 
 
-def resample(array, from_hz, to_hz, padtype=None):
+def resample(array, from_hz, to_hz, poly=False, padtype=None):
```
Comment (m-col) [inline pixels/signal.py]: Can we keep the default as True? Then the existing behaviour for other uses of this function don't need to be changed.

File: pixels/behaviours/base.py [hunk 3]
```diff
@@ -412,21 +412,51 @@ def process_lfp(self):
...
+
+            print(f"> Saving median subtracted & downsampled LFP to {output}")
+            # save in .npy format
+            np.save(
+                file=output,
+                arr=downsampled,
+                allow_pickle=True,
+            )
+            #downsampled = pd.DataFrame(downsampled)
+            #ioutils.write_hdf5(output, downsampled)
```
Comment (m-col) [inline pixels/behaviours/base.py:458]: Is npy here preferable over h5/dataframe? Just, no point in using a dataframe?

File: pixels/signal.py (hunk 2 above)
Comment (a-zmz) [reply pixels/signal.py]: sure can do

File: pixels/behaviours/base.py (hunk 3 above)
Comment (a-zmz) [reply pixels/behaviours/base.py:458]: currently i directly pass processed lfp data into k-means clustering calculation, which prefers ndarray. so if saving in df, we would need to convert ndarray to df and then later i would convert it back to ndarry again. seems repeating, so i commented out. would you prefer df?

Comment (m-col) [reply pixels/behaviours/base.py:458]: Nah either is fine

File: pixels/behaviours/base.py [hunk 4]
```diff
@@ -412,21 +412,53 @@ def process_lfp(self):
...
+            )
+            print(f"> Saving standard deviation (and their median) of each channel")
+            with open(sd, 'w') as fd:
+                json.dump(results, fd)
+
+            """
             if lag_end < 0:
                 data = data[:lag_end]
             if lag_start < 0:
                 data = data[- lag_start:]
```
Comment (m-col) [inline pixels/behaviours/base.py:449]: How come you've removed trimming the lag at the ends?

Comment (a-zmz) [reply pixels/behaviours/base.py:449]: oh right i commented it out for my own usage, should not have pushed it, will fix it

**Comment by m-col** [comment]
Thanks!

**Pull request merged**